### PR TITLE
Add docs for Azure Event Hubs changefeed sink v24.1

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-8052

This PR adds Azure Event Hubs as a supported sink for v24.1 docs. It includes:

- Sink list on the Overview page
- Step-by-step setup example on the [Changefeed Examples](https://deploy-preview-18414--cockroachdb-docs.netlify.app/docs/v24.1/changefeed-examples.html#create-a-changefeed-connected-to-an-azure-event-hubs-sink) page
- Details on the [Changefeed Sinks](https://deploy-preview-18414--cockroachdb-docs.netlify.app/docs/v24.1/changefeed-sinks.html#azure-event-hubs) page
- Parameters added to the [`CREATE CHANGEFEED`](https://deploy-preview-18414--cockroachdb-docs.netlify.app/docs/v24.1/create-changefeed.html#query-parameters) page

Refer to the links above for rendered previews of each page.

### CODE REVIEW

**Comment by kathancox** [comment]
@wenyihu6 friendly ping for a review here! Thanks!

**Comment by wenyihu6** [comment]
> @wenyihu6 friendly ping for a review here! Thanks!

Apologies for being late on the reviews here. I missed the GitHub email. I will review this PR today.

File: src/current/v24.1/changefeed-examples.md [hunk 1]
```diff
@@ -554,6 +555,62 @@ In this example, you'll set up a changefeed for a single-node cluster that is co
...
+    {% include_cached copy-clipboard.html %}
+    ~~~ sql
+    CREATE EXTERNAL CONNECTION eventhub AS 'azure-event-hub://{your-event-hubs-namespace}.servicebus.windows.net:9093?shared_access_key_name={policy-name}&shared_access_key={url-encoded key}';
+    ~~~
+
+1. Create the changefeed to your Event Hub:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ sql
+    CREATE CHANGEFEED FOR TABLE movr.rides, movr.users INTO 'external://eventhub';
```
[the remark this answers is not in the document]
Comment (kathancox) [reply src/current/v24.1/changefeed-examples.md:600]: They don't need to create an external connection, but I think it is probably a good to recommend that as part of the workflow. Not just for ease of use, but there are privileges connected with external connections that can help with fine-grained access control.

File: src/current/v24.1/changefeed-examples.md [hunk 2]
```diff
@@ -554,6 +555,62 @@ In this example, you'll set up a changefeed for a single-node cluster that is co
...
+1. Build the connection string to your Event Hub. You will need the following:
+    - The `azure-event-hub://` URI scheme.
+    - The **Host name** from your Event Hubs namespace overview page. It will be something like: `{your-event-hubs-namespace}.servicebus.windows.net`.
+    - The port `:9093` for the Kafka protocol.
+    - The `shared_access_key_name` parameter with the name of your policy.
+    - The `shared_access_key` parameter with your [**URL-encoded**](https://www.urlencoder.org/) primary key.
+
+    {% include {{ page.version.version }}/cdc/azure-event-hubs-uri.md %}
+
+    You can include the [`topic_name`]({% link {{ page.version.version }}/create-changefeed.md %}#topic-name-param) or [`topic_prefix`]({% link {{ page.version.version }}/create-changefeed.md %}#topic-prefix-param) parameters to create and name an Event Hub for emitted messages.
```
[the remark this answers is not in the document]
Comment (kathancox) [reply src/current/v24.1/changefeed-examples.md:587]: Using the Kafka documentation, and just generally from testing the example and other changefeed procedurals.

File: src/current/v24.1/create-changefeed.md [hunk 3]
```diff
@@ -140,16 +146,19 @@ Parameter          | <div style="width:100px">Sink Type</div>      | <div style=
...
-`sasl_enabled`     | [Kafka]({% link {{ page.version.version }}/changefeed-sinks.md %}#kafka), [Confluent Cloud]({% link {{ page.version.version }}/changefeed-sinks.md %}#confluent-cloud) | [`BOOL`]({% link {{ page.version.version }}/bool.md %})                 | If `true`, the authentication protocol can be set to SCRAM or PLAIN using the `sasl_mechanism` parameter. You must have `tls_enabled` set to `true` to use SASL. For Confluent Cloud sinks, `sasl_enabled` must be set to `true` if specified.<br><br> **Default:** `false`
+`sasl_enabled`     | [Azure Event Hubs]({% link {{ page.version.version }}/changefeed-sinks.md %}#azure-event-hubs), [Kafka]({% link {{ page.version.version }}/changefeed-sinks.md %}#kafka), [Confluent Cloud]({% link {{ page.version.version }}/changefeed-sinks.md %}#confluent-cloud) | [`BOOL`]({% link {{ page.version.version }}/bool.md %})                 | If `true`, the authentication protocol can be set to SCRAM or PLAIN using the `sasl_mechanism` parameter. You must have `tls_enabled` set to `true` to use SASL.<br><br>For Confluent Cloud and Azure Event Hubs sinks, this is set to `true` by default.<br><br> **Default:** `false`
```
Comment (Amruta-Ranade) [inline src/current/v24.1/create-changefeed.md:149]:
```suggestion
`sasl_enabled`     | [Azure Event Hubs]({% link {{ page.version.version }}/changefeed-sinks.md %}#azure-event-hubs), [Kafka]({% link {{ page.version.version }}/changefeed-sinks.md %}#kafka), [Confluent Cloud]({% link {{ page.version.version }}/changefeed-sinks.md %}#confluent-cloud) | [`BOOL`]({% link {{ page.version.version }}/bool.md %})                 | If `true`, the authentication protocol can be set to `SCRAM` or `PLAIN` using the `sasl_mechanism` parameter. You must have `tls_enabled` set to `true` to use SASL.<br><br>For Confluent Cloud and Azure Event Hubs sinks, this is set to `true` by default.<br><br> **Default:** `false`
```

Comment (Amruta-Ranade) [reply src/current/v24.1/create-changefeed.md:149]: Tiny change for consistency

**Review by Amruta-Ranade — APPROVED** [review]
LGTM!

**Review by wenyihu6 — APPROVED** [review]

**Comment by kathancox** [comment]
TFTRs @Amruta-Ranade @wenyihu6

**Pull request merged**